src/backend/distributed/utils/node_metadata.c

```diff
@@ -579,6 +611,81 @@ master_update_node(PG_FUNCTION_ARGS)
 }
 
 
+/* MetadataNodeList returns list of all synced metadata workers. */
+static List *
+SyncedMetadataNodeList(void)
+{
+	List *metadataNodeList = NIL;
+	List *activePrimaries = ActivePrimaryNodeList();
+	ListCell *workerNodeCell = NULL;
+
+	foreach(workerNodeCell, activePrimaries)
+	{
+		WorkerNode *workerNode = lfirst(workerNodeCell);
+		if (workerNode->hasMetadata && workerNode->metadataSynced)
+		{
+			metadataNodeList = lappend(metadataNodeList, workerNode);
+		}
+	}
+
+	return metadataNodeList;
+}
+
+
+/*
+ * WorkerListDelete removes the worker node with the given id
+ * from the list of given workers.
+ */
+static List *
+WorkerListDelete(List *workerList, uint32 nodeId)
+{
+	List *filteredWorkerList = NIL;
+	ListCell *workerCell = NULL;
+
+	foreach(workerCell, workerList)
+	{
+		WorkerNode *workerNode = lfirst(workerCell);
+		if (workerNode->nodeId != nodeId)
+		{
+			filteredWorkerList = lappend(filteredWorkerList, workerNode);
+		}
+	}
+
+	return filteredWorkerList;
+}
+
+
+/*
+ * SyncDistNodeEntryToNodes synchronizes the corresponding entry for
+ * the given workerNode in pg_dist_node metadata table of given
+ * metadataWorkers. If syncing to a node fails, pg_distnode.metadatasynced
+ * is set to false.
+ */
+static void
+SyncDistNodeEntryToNodes(WorkerNode *nodeToSync, List *metadataWorkers)
+{
+	ListCell *workerCell = NULL;
+	char *extensionOwner = CitusExtensionOwnerName();
+	char *nodeDeleteCommand = NodeDeleteCommand(nodeToSync->nodeId);
+	char *nodeInsertCommand = NodeListInsertCommand(list_make1(nodeToSync));
+	List *commandList = list_make2(nodeDeleteCommand, nodeInsertCommand);
+
+	foreach(workerCell, metadataWorkers)
+	{
+		WorkerNode *workerNode = lfirst(workerCell);
+		bool success =
+			SendOptionalCommandListToWorkerInTransaction(workerNode->workerName,
```

Comment:
Using this is not correct. It breaks transaction semantics. Instead I am creating a `SendOptionalCommandToWorkers()` with which is similar to `SendCommandToWorkers()` but uses SAVEPOINTs to rollback failures. Couldn't get it done today.